cscs-checks/tools/io/nco.py

```diff
@@ -69,15 +73,17 @@ def __init__(self, **kwargs):
         super().__init__('cdo_module_compat', **kwargs)
         self.descr = ('verifies compatibility with the CDO module')
         self.sourcesdir = None
-        self.executable = 'echo'
+        output_file = 'nco_cdo_check.out'
+        self.executable = 'echo > %s' % output_file
         self.sanity_patterns = sn.all([
-            sn.assert_not_found(r'.+', self.stdout),
+            sn.assert_not_found(r'.+', output_file),
```

Comment:
This is an identity, cos `output_file` will always be empty. The original intent of the test was to just check that the CDO can be loaded without conflict (see comment above). Since `module`  commands print to the standard error, it's sufficient to check that only this is empty or it does not contain the work `ERROR`.

@omlins Reading again the comment now, I wonder why this test should run on the compute nodes as well. If it was to run only on the login nodes, there would be no need for the `pre_run` and `echo` trick, would it?

Reply:
I think it should be run on the compute node to make sure that there it works, because there is where NCO and CDO must be run normally. The compute nodes and login nodes are supposed to be in sync, but sometimes differences may appear...

Reply:
Personally, I find it a bit redundant to run this on the compute nodes, cos it's sth completely related to the environment. Modules and homes are shared on the compute nodes, so they cannot be unsynced, unless there is some more fundamental problem which would come up earlier. Anyways, I've just proposed to run on the login nodes, just to make test simpler.